packages/base-types/src/request/index.ts

```diff
@@ -30,68 +30,78 @@ export interface Entity {
   name: string; // name of matched entity
   value: string; // inferred value of matched entity
   query?: string; // raw value of matched entity
+
+  /**
+   * @deprecated This value is no longer generated. Please remove all code relying on
+   * LUIS NLU to produce `verboseValue`
+   */
   verboseValue?: VerboseValue[]; // more detailed results from LUIS
 }
 
 export interface LabelRequestPayload {
   label?: string;
 }
-export interface BaseRequest<Payload = unknown> {
+export interface BaseRequest {
   type: string;
-  payload: Payload;
+  payload?: unknown;
   diagramID?: string; // particular topic to match against
 }
 
 export interface LaunchRequestPayload {
   persona?: string;
 }
 
-export interface LaunchRequest extends BaseRequest<LaunchRequestPayload | undefined> {
+export interface LaunchRequest extends BaseRequest {
   type: RequestType.LAUNCH;
+  payload?: LaunchRequestPayload;
 }
 
-export interface NoReplyRequest extends BaseRequest<undefined> {
+export interface NoReplyRequest extends Omit<BaseRequest, 'payload'> {
   type: RequestType.NO_REPLY;
 }
 
-export interface TextRequest extends BaseRequest<string> {
+export interface TextRequest extends BaseRequest {
   type: RequestType.TEXT;
+  payload: string;
 }
 
-interface ActionAndLabelRequestPayload extends ActionPayload, LabelRequestPayload {}
+interface ActionAndLabelRequestPayload extends ActionPayload, LabelRequestPayload { }
 
 export interface IntentRequestPayload extends ActionAndLabelRequestPayload {
-  query: string; // original text input
   intent: { name: string }; // matched intent name
-  entities: Entity[]; // entities matches - multiple of the same entity name may be matched
+  entities?: Entity[]; // entities matches - multiple of the same entity name may be matched
   confidence?: number; // 0-1 confidence of match;
+  query?: string; // original text input
   data?: Record<string, unknown>;
```

Comment:
According to Tyler H, the `query` and `entities` can potentially be optional, so the `@voiceflow/dtos` type marks these as option. Here, we mark the properties as optional to keep alignment.